packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -70,217 +73,375 @@ fn get_pkg_path(name: &str, pkg_type: PackageType) -> String {
-            .eval(wrapped_code)
-            .map_err(|e| Error::PackageImport(path.clone(), e.into()))?;
-        let args = mv8::Values::from_vec(vec![
-            embedded.hash_util.clone(),
-            embedded.hash_stdlib.clone(),
-        ]);
-
-        let fns: mv8::Array<'_> = pkg
-            .call(args)
-            .map_err(|e| Error::PackageImport(path.clone(), e.into()))?;
-        if fns.len() != 3 {
-            return Err(Error::PackageImport(path.clone(), "Stray return".into()));
+        // Double braces like `{{` are Rust's escape string for a single literal `{`.
+        let wrapped_code = new_js_string(
+            scope,
+            &format!(
+                "((hash_util, hash_stdlib) => {{
+                    {code}
+                    return [
+                        typeof start_experiment === 'undefined' ? undefined : start_experiment,
+                        typeof start_sim === 'undefined' ? undefined : start_sim,
+                        typeof run_task === 'undefined' ? undefined : run_task
+                    ]
+                }})",
+            ),
+        )?;
+
+        let pkg: v8::Local<'_, v8::Function> = {
+            let pkg = v8::Script::compile(scope, wrapped_code, None).ok_or_else(|| {
+                JSRunnerError::V8("Could not create wrapped code scipt".to_string())
+            })?;
+            let pkg = pkg.run(scope).ok_or_else(|| {
+                JSRunnerError::V8("Could not execute wrapped code script".to_string())
+            })?;
+
+            pkg.try_into().map_err(|e| {
+                JSRunnerError::V8(format!("Could not convert wrapped code to Function: {e}"))
+            })?
+        };
+
+        let args = &[embedded.hash_util, embedded.hash_stdlib];
+        let fns: v8::Local<'_, v8::Array> = {
+            let global_context = context.global(scope);
+            let fns = pkg
+                .call(scope, global_context.into(), args)
+                .ok_or_else(|| {
+                    JSRunnerError::PackageImport(
+                        path.clone(),
+                        "Could not run wrapped code function".to_string(),
+                    )
+                })?;
+
+            fns.try_into().map_err(|e| {
+                JSRunnerError::V8(format!(
+                    "Could not convert wrapped code return value to Array: {e}"
+                ))
+            })?
+        };
+
+        if fns.length() != 3 {
+            return Err(JSRunnerError::PackageImport(
+                path.clone(),
+                "Stray return".into(),
+            ));
         }
 
         // Validate returned array.
         let fn_names = ["start_experiment", "start_sim", "run_task"];
-        for (elem, fn_name) in fns.clone().elements().zip(fn_names) {
-            let elem: mv8::Value<'_> = elem.map_err(|e| {
-                Error::PackageImport(path.clone(), format!("Couldn't index array: {:?}", e))
+        for (i, fn_name) in (0..fns.length()).zip(fn_names) {
```

Comment:
We are trying to get rid of single-letter variables (also in closures). The plan is to migrate `i` to `xx_idx`, so in this case it would be `fn_idx`. We currently don't change already existing code until we migrate completely, but newly introduced variables should follow this rule. This also applies to `map_err(|e|)` -> `map_err(|err|)`.

Reply:
Addressed in https://github.com/hashintel/hash/pull/414/commits/02af15786b564ada4f9992194281ff0a3281afb9